compiler/modules/front/codeGen.bal

```diff
@@ -777,9 +777,23 @@ function codeGenAssignStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environ
     if lValue is s:VarRefExpr {
         return codeGenAssignToVar(cx, startBlock, env, lValue.varName, expr);
     }
-    else {
+    else if lValue is s:MemberAccessExpr {
```

Comment:
Don't make the wildcard the last one, so you get better type checking

Reply:
Fixed